Assets/Scripts/Singleton.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Singleton<T> : MonoBehaviour where T : MonoBehaviour
```

Comment:
you should cite where you copy and pasted code from

Reply:
Also changing it from the consts wasn't really necessary and makes the code harder to read. Magic strings for logging can make the code more readable. Consts are best used when you have something that is never going to change and is going to be reused or could be reused. I.e. dictionary keys is a good example. This is too extreme on the make everything const scale.